logprep/connector/file/input.py

```diff
@@ -0,0 +1,346 @@
+                baseline_crc32, baseline_fingerprint_size = self._calc_file_fingerprint(file)
+                self._fileinfo_util.add_fingerprint(
+                    file_name, baseline_crc32, baseline_fingerprint_size
+                )
+
+            baseline_fingerprint_size = self._fileinfo_util.get_fingerprint_size(file_name)
+            check_crc32, _ = self._calc_file_fingerprint(file, baseline_fingerprint_size)
+            if not self._fileinfo_util.check_fingerprint(file_name, check_crc32):
+                # if the fingerprint in check_crc32 is not the same a non-appending
+                # file change happened, create new baseline with potential new fingerprint_size
+                new_check_crc32, new_fingerprint_size = self._calc_file_fingerprint(file)
+                self._fileinfo_util.add_fingerprint(
+                    file_name, new_check_crc32, new_fingerprint_size
+                )
+                self._fileinfo_util.add_offset(file_name, 0)
+
+            while True:
+                if self._fileinfo_util.get_offset(file_name):
+                    # if the file offset already exists, continue from there
+                    file.seek(self._fileinfo_util.get_offset(file_name))
+
+                line = file.readline()
+                # add new file offset after line of file was read
+                self._fileinfo_util.add_offset(file_name, file.tell())
+                if line != "":
+                    if line.endswith("\n"):
+                        message = self._line_to_dict(line)
+                        if message:
+                            self._messages.put(message)
+                else:
+                    break
+
+    def _line_to_dict(self, input_line: str) -> dict:
+        """Takes an input string and turns it into a dict without any parsing or formatting.
+        Only thing it does additionally is stripping the new lines away."""
+        input_line = input_line.rstrip("\n")
+        if (isinstance(input_line, str)) and (len(input_line) > 0):
+            return {"message": input_line}
+        return ""
+
+    def _get_event(self, timeout: float) -> tuple:
+        """Returns the first message from the threadsafe queue"""
+        try:
+            message = self._messages.get(timeout=timeout)
+            raw_message = str(message).encode("utf8")
+            return message, raw_message
+        except queue.Empty:
+            return None, None
+
+    def setup(self):
+        """Creates and starts the Thread that continously monitors the given logfile.
+        Right now this input connector is only started in the first process.
+        It needs the class attribute pipeline_index before running setup in Pipeline
+        Initiation
+
+        TODO (optional): when processing multiple files, map the threads
+        equally-distributed to each process"""
+        # create the event that will allow to kill the thread
+        self.stop_flag = threading.Event()
```

Comment:
This would require an `__init__` function? I can do that but especially regarding the Spawning of the Connectors I recognized some odd behaviour I don't understand. Running the Logprep in this branch with `processor_count: 2` with print statements in __init__ and setup function:

```
(logprepenv) user@elmo:~$ logprep ownpipe/own_pipeline.yml 
2023-01-22 20:00:44,834 Logprep INFO    : Log level set to 'INFO'
2023-01-22 20:00:44,840 Logprep INFO    : python version:          3.9.5
2023-01-22 20:00:44,840 Logprep INFO    : logprep version:         4.0.0+76.gf8e5b38.dirty
2023-01-22 20:00:44,840 Logprep INFO    : configuration version:   1, file://ownpipe/own_pipeline.yml
Running  __init__ of FileInput
Instance ID in __init__: 140075310129056
Running  __init__ of FileInput
Instance ID in __init__: 140075309734016
2023-01-22 20:00:44,855 Logprep INFO    : Metric tracking is disabled via config
Running  __init__ of FileInput
Instance ID in __init__: 140075309735456
2023-01-22 20:00:44,871 Logprep INFO    : Created new pipeline
2023-01-22 20:00:44,874 Logprep INFO    : Created new pipeline
2023-01-22 20:00:44,877 Logprep INFO    : Startup complete
Running  __init__ of FileInput
Instance ID in __init__: 140075309733728
Running setup of FileInput
Instance ID in setup: 140075309733728
Running  __init__ of FileInput
Instance ID in __init__: 140075309733728
Running setup of FileInput
Instance ID in setup: 140075309733728
```

Do your see that the setup function is only as often called it was defined in processor_count but the __init__ function itself is called more often and with at least one more instance as defined in processor_count. It seems when running setup for each Pipeline, the same instance is used in the different address space of each processor (at least that is my explanation of this behaviour. For this reason I removed the __init__ function again, I did have added in first place.

Reply:
actually it also works when the `stop_flag` event is defined in __init__ but as I mentionned I tried to avoid areas I don't understand.